src/Guild.js

```diff
@@ -25,6 +25,35 @@ class Guild {
     autoBind(this);
   }
 
+  guildRoleCreate (roleId) {
+    this.roles[this.roles.name] = roleId;
+  }
+
+  guildGetKeyByValue (value) {
+    return Object.keys(this.roles).find(key => this.roles[key] === value);
+  }
+
+  guildUpdateRoleName (roleId) {
+    const oldKey = this.guildGetKeyByValue(roles.id);
+
+    //role for roleId does not exist. nothing to update
+    if(!this.isRole(oldKey)) {
```

Comment:
"If the name already exists, return."

At the moment, you are checking if the Role name DOESN'T exist. You should also be using the Role name you got from the event, not the `oldKey`.

Reply:
Offcourse! thanks. Fixed it now